docs/docs/how-to/plugins-and-themes/adding-gatsby-image-support.md

```diff
@@ -0,0 +1,261 @@
+Support for these are available out of the box in `gatsby-transformer-sharp`, so if your plugin downloads images and processes them locally then your users can use the [`gatsbyImageData` resolver](https://www.gatsbyjs.com/docs/how-to/images-and-media/using-gatsby-plugin-image#dynamic-images). However, if your CDN can deliver images of multiple sizes with a URL-based API, then the plugin includes a toolkit to allow you to give your users the same great experience without needing to download the images locally. It also allows you to create components that display these images dynamically at runtime, without needing to add a GraphQL resolver.
+
+## Adding a `gatsbyImageData` GraphQL resolver
+
+You can give your users the best experience by adding a `gatsbyImageData` resolver to your image nodes. This allows you to generate low-resolution or traced SVG images as inline data URIs, so your users can have blurred placeholders. You can also calculate the image's dominant color for an alternative placeholder. These are the same placeholders that are included with `gatsby-transformer-sharp`, and will give the best experience for your users. If you are able to deliver these directly from your CMS or other data source then this is ideal, but otherwise you can use helper functions included in `gatsby-plugin-sharp`.
+
+There are three steps to add a basic `gatsbyImageData` resolver:
+
+1. [Create the `generateImageSource` function](#create-the-generateimagesource-function)
+2. [Create the resolver function](#create-the-resolver-function)
+3. [Add the resolver](#add-the-resolver)
+
+### Create the `generateImageSource` function
+
+The `generateImageSource` function is where you generate your image URLs. The image plugin calculates which sizes and formats are needed, according to the format, size and breakpoints requested by the user. For each of these, your function is passed the base URL, width, height and format (i.e. the image filetytpe), as well as any custom options that your plugin needs. You then return the generated URL. The returned object also includes width, height and format. This means you can return a different value from the one requested. For example, if the function requests an unsupported format or size, you can return a different one which will be used instead.
+
+```js:title=gatsby-source-example/gatsby-node.js
+// In this example we use a custom `quality` option
+const generateImageSource = (baseURL, width, height, format, fit, options) => {
+  const src = `https://myexampleimagehost.com/${baseURL}?w=${width}&h=${height}&fmt=${format}&q=${options.quality}`
+  return { src, width, height, format }
+}
+```
+
+### Create the resolver function
+
+You then can use the function created in the previous step to build your resolver function. It can be an async function, and it should return the value from `generateImageData`. An example resolver could look like this:
```

Comment:
```suggestion
You can then use the function created in the previous step to build your resolver function. It can be an async function, and it should return the value from `generateImageData`. An example resolver could look like this:
```